test/e2e/suite/serving/cainjector.go

```diff
@@ -150,21 +149,19 @@ var _ = framework.CertManagerDescribe("CA Injector", func() {
 				if test.disabled != "" {
 					Skip(test.disabled)
 				}
-				injectable, cert, _ := generalSetup(test.makeInjectable("changed"))
-
-				By("grabbing the latest copy of the cert")
-				Expect(f.CRClient.Get(context.Background(), types.NamespacedName{Name: cert.Name, Namespace: cert.Namespace}, &cert)).To(Succeed())
+				injectable, cert := generalSetup(test.makeInjectable("changed"))
 
 				By("changing the name of the corresponding secret in the cert")
-				secretName := types.NamespacedName{Name: cert.Spec.SecretName, Namespace: f.Namespace.Name}
+				cert = cert.DeepCopy() // DeepCopy before updating
```

Comment:
Ah, was that the cause of the intermittent failure? 
Did you consider doing the DeepCopy in the `WaitForCertificate...` helper functions, to help avoid this same bug happening elsewhere?

Reply:
This could be related, but I fear that there are multiple errors causing failures.
I would like to keep the `cert = cert.DeepCopy()` visible in the code, so that we can more easily see when DeepCopies are needed.